lib/clickhousex/protocol.ex

```diff
@@ -74,84 +68,88 @@ defmodule Clickhousex.Protocol do
     end
   end
 
-  @doc false
-  @spec reconnect(new_opts :: Keyword.t(), state) :: {:ok, state}
-  def reconnect(new_opts, state) do
-    with :ok <- disconnect("Reconnecting", state),
-         do: connect(new_opts)
+  # TODO find a cleaner way
```

Comment:
I'm not sure if this is the right approach. It might solve the noisy logs problem but it doesn't seem clean. I'd prefer if dbconnection was doing the reconnect logic.

Reply:
Maybe it would be a good idea to get some feedback from the maintainers of DBConnection on this?